jupyterhub/singleuser.py

```diff
@@ -38,14 +48,17 @@
     TraitError,
 )
 
-from notebook.notebookapp import (
-    NotebookApp,
-    aliases as notebook_aliases,
-    flags as notebook_flags,
+app_module = importlib.import_module(app_name)
```

Comment:
There's a new recommended pattern for this (which I think is bizarre, personally): https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly

Reply:
@blink1073 that's only for importing source files right? We're importing a module directly here...

Reply:
Ah, I see what you mean, thanks!